### PR TITLE
Popover: enable auto-updating every animation frame

### LINKED ISSUE #42725: Site editor: Block settings dropdown remains in place when scrolling
### Step-by-step reproduction instructions

1. Ensure you're editing a template with enough content that a scrollbar displays
2. Select a block and open the block settings dropdown
3. Scroll the content

Expected: the block settings dropdown scrolls with the content (matching the toolbar's position)
Actual: the block settings dropdown remains in place while scrolling

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/677833/181190083-99f4de96-85bf-4527-8361-dd4d16c29652.mp4

### Environment info

Latest trunk

Reproduced using Mac OS / Brave

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by ciampo**
## Dev note

See the dev note in https://github.com/WordPress/gutenberg/pull/44195

## What?

Fixes #42725

This PR changes the way `floating-ui` auto-updates the popover's position by enabling the `animationFrame` option. This means that the popover position is recalculated every animation frame (see [the docs](https://floating-ui.com/docs/autoupdate#animationframe) for more details).

This has a few benefits:

 - The "second gen" popovers are updated correctly without needing to add additional event listeners (see #42725)
 - The `__unstableObserveElement` prop (and the associated `MutationObserver`) are not necessary anymore, and can be removed
 - The `scroll` event listener added to the `iframe` is not necessary anymore, and can be removed
 - Popovers tracking anchors across different frames (i.e the block toolbar in the site editor) are less laggy, since they are updated every animation frame

## Why?

Currently, the `Popover` element adds some custom event listeners to the anchor and its `onwerDocument` when the anchor is in a different document from the popover (i.e. in an iframe, which is the case for the block toolbar in the site editor).

These events don't cover all edge cases, and are therefore prone to causing regressions when other parts of the `Popover` are updated.

## How?

By setting the `animationFrame` option to `true` when calling the `autoUpdate` function from `floating-ui`

## What is not included in this PR

The changes in this PR were inspired by the work done in https://github.com/WordPress/gutenberg/pull/43335, but I've deliberately decided to keep this PR as small as possible to make it easier to review.

I plan on iterate on `Popover` improvements in https://github.com/WordPress/gutenberg/pull/43335, where I'm exploring / going to explore the following ideas:

- calling `autoUpdate` only as part of the `useFloating` function call (and not in the `useLayoutEffect` hook, like it is at the moment)
- leverage more internal state and references, in order to have the `Popover` component update correctly and more frequently in more scenarios (e.g. prop change, ref changes...)
- update the `reference` within the same `useLayoutEffect` where the reference itself is computed
- use `ResizeObserver` instead of listening to the `resize` event
- see if it's possible to simplify computing the `ownerDocument`

## Testing Instructions

- In the site editor, select a block. From the the block toolbar, open any submenu (like the the settings menu) that causes a second popover to open
- scroll the document, and make sure that both popovers update their position correctly

Check that the remaining popovers behave as expected, without any regressions..

## Screenshots or screencast 

https://user-images.githubusercontent.com/1083581/186689790-e553fb3e-8c44-4ded-bdc2-8b753304a147.mp4

### CODE REVIEW

**Review by talldan — APPROVED** [review]
LGTM! Thanks for exploring an alternative fix.

A note about performance, in testing the toolbar/popover position is laggier when scrolling if a toolbar dropdown (like the block settings menu) is open, but I don't notice any difference if toolbar dropdowns are closed.

I think it's more important to fix the bug right now, and then look at performance separately, so it'd be good to move ahead with merging this.

**Comment by ciampo** [comment]
Just rebased this PR, will merge once CI is ✅ 

Regarding performance, I didn't really spot increased lagginess, but I really didn't do much testing on that, as I focused on fixing the bugs.

Once this PR gets merged, I should be able to work on https://github.com/WordPress/gutenberg/pull/43335, which hopefully may bring some perf improvements 🤞 although the "real" fix will be to improve cross-document support directly in `floating-ui` (https://github.com/floating-ui/floating-ui/issues/1869)

**Comment by ciampo** [comment]
This PR seems to have broken the [draggable block e2e tests](https://github.com/WordPress/gutenberg/blob/c4bdc602cce6b885eca90a538a412d7175039689/packages/e2e-tests/specs/editor/various/draggable-block.test.js), I'll see if I can make the test work — otherwise I will open a PR that will  partially undo the changes from this PR

**Comment by ciampo** [comment]
#43734

File: packages/components/src/popover/index.js
Comment (youknowriad) [inline packages/components/src/popover/index.js:378]: I noticed that you did some follow-up work to this, so I was wondering if this is still necessary. The reason I'm asking is because this line is a source of a large impact on performance degradation potentially.

It now happens in the "select" and "inserter open" metrics after this PR https://github.com/WordPress/gutenberg/pull/42722 but it can also happen I guess for other interactions when there's a big list of popovers.

So can we remove it or is it still necessary? Are there alternatives?

Comment (kevin940726) [reply packages/components/src/popover/index.js:378]: I guess this can be opt-in (or opt-out). However, I think it's generally a bad practice to render a large list of popovers at the same time. The performance regression in #42722 is a bit extreme that we render 1000 empty paragraphs in the test, but I do think we can refactor it in a follow-up to improve this.

From the [floating-ui](https://floating-ui.com/docs/autoUpdate#animationframe) doc:

> This is optimized for performance but can still be costly.

> This option should be used sparingly...

I guess it makes sense to make it an opt-in in the next major release?

Comment (ciampo) [reply packages/components/src/popover/index.js:378]: This was introduced mainly to fix https://github.com/WordPress/gutenberg/issues/42725, I expect that bug (and any other nested popover) to happen again if we undo this change — maybe it could be just enabled for the block toolbar?

While I agree that calling the `getBoundingClientRect` function every animation frame is not ideal, I would also argue that the main reason while this can be a noticeable regression is that we're rendering hundreds of popovers at the same time.

Comment (youknowriad) [reply packages/components/src/popover/index.js:378]: It turned out that the main issue for the performance regression is not this, so I'm ok not doing anything here and keeping it as but we should keep an eye on this any way as mentioned on their docs.

Comment (ciampo) [reply packages/components/src/popover/index.js:378]: Sure, makes sense

**Pull request merged**
